src/invoke.cpp

```diff
@@ -2872,20 +2906,62 @@ void fcall_urdb_get(lk::invoke_t &cxt)
+		}
+		if (!applydiurnalschedule(cxt, "cr_sched", rate.Unused.CoincidentSchedule)) return;
+		if (!copy_mat(cxt, "cr_tou_mat", rate.Unused.CoincidentRateStructure)) return;
+
+		cxt.result().hash_item("energyattrs").assign(rate.Unused.EnergyAttrs);
+		cxt.result().hash_item("demandattrs").assign(rate.Unused.DemandAttrs);
+		cxt.result().hash_item("fixedattrs").assign(rate.Unused.FixedAttrs);
+
 		// URLs
 		cxt.result().hash_item("rateurl").assign(rate.Header.RateURL);
 		cxt.result().hash_item("jsonurl").assign(rate.Header.JSONURL);
-
-		// net metering
-		if (rate.NetMetering)
-			cxt.result().hash_item("enable_net_metering").assign(1.0);
-		else
-			cxt.result().hash_item("enable_net_metering").assign(0.0);
-
+		
+		// metering option
+		// "Net Metering", "Net Billing Instantaneous", "Net Billing Hourly", or "Buy All Sell All"
+		if (rate.DgRules == "Net Metering")
+			cxt.result().hash_item("metering_option").assign(0.0);
+		else if (rate.DgRules == "Net Billing Instantaneous")
+			cxt.result().hash_item("metering_option").assign(2.0);
+		else if (rate.DgRules == "Net Billing Hourly")
+			cxt.result().hash_item("metering_option").assign(2.0);
+		else if (rate.DgRules == "Buy All Sell All")
+			cxt.result().hash_item("metering_option").assign(4.0);
+        else // set to default Net Energy Metering
+        {
+            cxt.result().hash_item("metering_option").assign(0.0);
+            rate_notes.append(wxString::Format("Metering option not provided with rate data.\n"));
+        }
+		
 		// fixed charges
-		cxt.result().hash_item("monthly_fixed_charge").assign(rate.FixedMonthlyCharge);
-		cxt.result().hash_item("monthly_min_charge").assign(rate.MinMonthlyCharge);
-		cxt.result().hash_item("annual_min_charge").assign(rate.MinAnnualCharge);
+		//  "$/day", "$/month" or "$/year" TO DO handle $/day and $/year
+        double fixed_charges = rate.FixedChargeFirstMeter + rate.FixedChargeAddlMeter;
+        cxt.result().hash_item("monthly_fixed_charge").assign(0.0);
+        if ((rate.FixedChargeUnits) == "$/month")
+            cxt.result().hash_item("monthly_fixed_charge").assign(fixed_charges);
+        else if ( fixed_charges > 0 )
+            rate_notes.append(wxString::Format("SAM does not model fixed charge rate of %f with %s units.\n", fixed_charges, rate.FixedChargeUnits));
+
+		// minimum charges
+		// "$/day", "$/month" or "$/year" TO DO handle $/day
+        cxt.result().hash_item("monthly_min_charge").assign(0.0);
+        cxt.result().hash_item("annual_min_charge").assign(0.0);
+        if (rate.MinChargeUnits == "$/month")
+			cxt.result().hash_item("monthly_min_charge").assign(rate.MinCharge);
+		else if ( rate.MinChargeUnits == "$/year")
+			cxt.result().hash_item("annual_min_charge").assign(rate.MinCharge);
+        else if (rate.MinCharge > 0 )
+            rate_notes.append(wxString::Format("SAM does not model minimum charge rate of %f with %s units.\n", rate.MinCharge, rate.MinChargeUnits));
```

Comment:
Similar problem with this rate: https://en.openei.org/apps/IURDB/rate/view/5c817fbf7bece2299eceb397 and the error message for minimum charges: https://en.openei.org/apps/IURDB/rate/view/5c817fbf7bece2299eceb397